src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -621,45 +621,37 @@ void validateForByRequestIdModeApproved() {
     SnapshotRequestContentsModel snapshotRequestContentsModel =
         makeByRequestIdContentsModel(snapshotAccessRequestId);
     SnapshotAccessRequestModel accessRequestResponse =
-        new SnapshotAccessRequestModel(
-            null,
-            null,
-            null,
-            null,
-            null,
-            "email@a.com",
-            null,
-            null,
-            SnapshotAccessRequestStatus.APPROVED,
-            null,
-            null);
+        SnapshotBuilderTestData.createAccessRequest();
     when(snapshotRequestDao.getById(snapshotAccessRequestId)).thenReturn(accessRequestResponse);
 
     assertDoesNotThrow(() -> service.validateForByRequestIdMode(snapshotRequestContentsModel));
   }
 
+  static SnapshotAccessRequestModel createAccessRequestWithFlightid() {
+    return new SnapshotAccessRequestModel(
+        null,
+        null,
+        null,
+        null,
+        null,
+        "email@a.com",
+        null,
+        null,
+        SnapshotAccessRequestStatus.APPROVED,
+        null,
+        "flightId");
+  }
```

Comment:
love the changes you made to tests to use the new method SnapshotBuilderTestData.createAccessRequest(). Here, can you use it too and then just set the flightId?

Reply:
Since `SnapshotAccessRequestModel` is a `record` its fields are immutable and can't be set after creation. One pattern I've used is to add methods to the record that return a new object with a changed field, like:
```
  public SnapshotAccessRequestModel withFlightid(String flightid) {
    return new SnapshotAccessRequestModel(
        id,
        snapshotName,
        snapshotResearchPurpose,
        sourceSnapshotId,
        snapshotSpecification,
        createdBy,
        createdDate,
        statusUpdatedDate,
        status,
        createdSnapshotId,
        flightid);
  }
```

I could do that here, but this case only comes up in testing (so far) and it's only used in a couple places, so it didn't seem worth adding non-test code to support this case.